lib/semian/circuit_breaker.rb

```diff
@@ -30,9 +32,7 @@ def acquire
     end
 
     def request_allowed?
-      return true if closed?
-      half_open if error_timeout_expired?
-      !open?
+      closed? || half_open? || (open? && error_timeout_expired?)
```

Comment:
`error_timeout_expired?` uses `@errors.last` which is calling `.last` on an array.  That is probably safe to do in ruby due to the vm lock, but we might want to have a `@last` instance variable in the SlidingWindow class if you want it to be accessed in a thread-safe way for jruby.